### ISSUE TITLE
[JXR-154] xref-test package summary lists also classes from main source directories

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-jxr#265]
**[Andreas Sewe](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=asewe)** opened **[JXR-154](https://issues.apache.org/jira/browse/JXR-154?redirect=false)** and commented

This is a regression from 3.0.0:

The `xref-test` report lists not only classes from the test source directories, but also from the main source directories in its `package-summary.html`. This is not only unexpected, but the respective links to the actual sources are also dead.

Using `git bisect`, I was able to narrow this bug (introduced between jxr-3.0.0 and 82193ede1e036a0e66e1591295f2311982a17c73) down to the following commit:

```
b4117e7ef0795eed02e2ace3c9c4c310f5391aac is the first bad commit
commit b4117e7ef0795eed02e2ace3c9c4c310f5391aac
Author: rfscholte <rfscholte@apache.org>
Date:   Fri Jul 5 00:11:29 2019 +0200    [JXR-142] Switch to JSR330 Dependency Injection,
    [JXR-144] Require Maven 3.1.0:040000 040000 9bd9a415b2e3d94406911ca475603de67b504be9 dc78032a3f63c806041f360da96e9468c955f421 M	maven-jxr-plugin
:040000 040000 a9bbe183fcbc2b22afae203be4483fd9c1dd2a1b c15f3e82de0050ffa3834c0ff076fa8a6b57c31a M	maven-jxr
:100644 100644 e61a9f56ba6ea12165b72967873f72ed7611f894 be2f7ecc125a573c7316c048dc521c40c60aa5f1 M	pom.xml
bisect run success 
```

using the attached patch [^it.patch] to the most basic integration test, `simple-project`.

---

**Affects:** 3.1.1

**Attachments:**
- [it.patch](https://issues.apache.org/jira/secure/attachment/13003966/it.patch) (_1.11 kB_)

**Issue Links:**
- [JXR-160](https://issues.apache.org/jira/browse/JXR-160) 3.1.0 not in Maven Central
 (_**"causes"**_)
- [JXR-142](https://issues.apache.org/jira/browse/JXR-142) Switch to CDI
 (_**"is caused by"**_)

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

Yes, we need to reorganize code. The intention to switch to components has to do with JXR-145. That way it will be easier to inject a GrooveCodeTransformer or other language.
There was no official API, but I knew some tools use JXR as well. But that just doesn't fit anymore.
I also discovered that test code does not link back to main code (as external jxr pages). That would be a nice improvement too.

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

Fixed in [473e1b9bd9d38c703e48e69bcd17d126d1d9863d](https://gitbox.apache.org/repos/asf?p=maven-jxr.git;a=commit;h=473e1b9bd9d38c703e48e69bcd17d126d1d9863d)
